### PR TITLE
Support unions in GraphQL

### LINKED ISSUE #251: Union types support for GraphQL
Just finished playing with this library and noticed that union types for Graphql are not supported.
Is this something on a roadmap or no plans to support them in general?

### PR DESCRIPTION
**Pull request by serras**
Fixes #251

### CODE REVIEW

File: graphql/src/Mu/GraphQL/Query/Parse.hs [hunk 1]
```diff
@@ -22,8 +22,8 @@ import           Data.Int                    (Int32)
 import           Data.List                   (find)
 import           Data.Maybe
 import           Data.Proxy
-import           Data.Scientific             (Scientific, floatingOrInteger, fromFloatDigits)
 import           Data.SOP.NS
+import           Data.Scientific             (Scientific, floatingOrInteger, fromFloatDigits)
```
Comment (kutyel) [inline graphql/src/Mu/GraphQL/Query/Parse.hs:26]: I think you haven't run stylish-haskell here? 🤔

File: graphql/src/Mu/GraphQL/Query/Parse.hs [hunk 2]
```diff
@@ -263,59 +267,91 @@ constToValue (GQL.ConstObject n)
...
+      go (GQL.FragmentSpreadSelection (GQL.FragmentSpread nm dirs _) : ss)
+        | Just fr <- HM.lookup nm frmap
+        = if not (any (shouldSkip vmap) dirs) && not (any (shouldSkip vmap) $ fdDirectives fr)
+            then (++) <$> go (fdSelectionSet fr) <*> go ss
+            else go ss
+        | otherwise  -- the fragment definition was not found
+        = throwError $ "fragment '" <> nm <> "' was not found"
+      go (_ : _)  -- Inline fragments are not yet supported
+        = throwError "inline fragments are not (yet) supported"
+      -- fieldToMethod :: GQL.Field -> f (Maybe (OneMethodQuery p ('Service sname methods)))
```
Comment (kutyel) [inline graphql/src/Mu/GraphQL/Query/Parse.hs:328]: Can we uncomment this line?

File: servant/server/src/Mu/Servant/Server.hs [hunk 3]
```diff
@@ -47,10 +47,10 @@ import qualified Data.ByteString.Lazy.UTF8 as LB8
 import           Data.Conduit.Internal     (ConduitT (..), Pipe (..))
 import           Data.Kind
 import           Data.Swagger              (Swagger, ToSchema (..))
-import           Generics.Generic.Aeson
 import           GHC.Generics
 import           GHC.TypeLits
 import           GHC.Types                 (Any)
+import           Generics.Generic.Aeson
```
Comment (kutyel) [inline servant/server/src/Mu/Servant/Server.hs:53]: stylishhhhh 💅🏻

File: core/rpc/src/Mu/Server.hs [hunk 4]
```diff
@@ -151,17 +154,39 @@ data ServerT (chn :: ServiceChain snm) (info :: Type)
...
                (s :: [Service snm mnm anm (TypeRef snm)])
                (m :: Type -> Type) (hs :: [[Type]]) where
   S0 :: ServicesT chn info '[] m '[]
-  (:<&>:) :: HandlersT chn info (MappingRight chn sname) methods m hs
+  (:<&>:) :: ServiceT chn info svc m hs
           -> ServicesT chn info rest m hss
-          -> ServicesT chn info ('Service sname methods ': rest) m (hs ': hss)
+          -> ServicesT chn info (svc ': rest) m (hs ': hss)
+
+type family InUnion (x :: k) (xs :: [k]) :: Constraint where
```
Comment (kutyel) [inline core/rpc/src/Mu/Server.hs:169]: 👨‍👦‍👦 ❤️

**Review by kutyel — APPROVED** [review]
Awesome work! 🙌🏻

File: graphql/src/Mu/GraphQL/Query/Parse.hs (hunk 2 above)
Comment (serras) [reply graphql/src/Mu/GraphQL/Query/Parse.hs:328]: Not really, due to GADT matching.

**Pull request merged**